tests/conftest.py

```diff
@@ -8,3 +10,12 @@ def global_config_fixture():
     global_config.reset()
     yield global_config
     global_config.reset()
+
+
+@pytest.fixture(autouse=True)
+def fail_on_user_warnings():
```

Comment:
I'm in two minds about this fixture &mdash; it does catch this issue and prevent any similar issues in future, but I don't like having to suppress the one deprecation warning that Pydantic appears to have accidentally left as a UserWarning. LMK if you'd rather leave it out, it's a separate commit so easy to revert.

Reply:
I would rather not optimise early here as it might get difficult/annoying when we move Hera's core code to Pydantic V2

Reply:
Sure thing. Removed the fixture.